backend/src/middleware/auth.ts

```diff
@@ -1,9 +1,40 @@
 import dotenv from 'dotenv';
 import { auth } from 'express-oauth2-jwt-bearer';
+import { Request, Response, NextFunction } from 'express';
+import jwt from 'jsonwebtoken';
+import User from '../models/user';
+
 dotenv.config();
 
 export const jwtCheck = auth({
   audience: process.env.AUTH0_AUDIENCE,
   issuerBaseURL: process.env.AUTH0_IS_USER_BASE_URL,
   tokenSigningAlg: 'RS256',
 });
+
+export const jwtParse = async (req: Request, res: Response, next: NextFunction) => {
+  const { authorization } = req.headers;
+  if (!authorization || !authorization.startsWith('Bearer ')) {
+    return res.sendStatus(401);
+  }
+  const token = authorization.split(' ')[1];
+
+  try {
+    const decoded = jwt.decode(token);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/18#issuecomment-2378818894>#d6f619</a><span> - 09/27/2024, 09:14 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The jwt.decode() method is used to parse the JWT token, which doesn't verify the token's signature. This could lead to accepting tampered tokens.
</div>
<div id="fix">
<b>Fix:</b> Replace jwt.decode() with jwt.verify() to ensure the token's signature is validated. Use the appropriate secret or public key for verification.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -20,7 +20,7 @@ export const jwtParse = async (req: Request, res: Response, next: NextFunction)
    const token = authorization.split(' ')[1];

    try {
 -    const decoded = jwt.decode(token);
 +    const decoded = jwt.verify(token, process.env.JWT_SECRET);
      if (!decoded) {
        return res.sendStatus(401);
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
good point